rdtools/availability.py

```diff
@@ -0,0 +1,565 @@
+"""
+Functions for detecting and quantifying production loss from photovoltaic
+system downtime events.
+"""
+
+import rdtools
+
+import pandas as pd
+import numpy as np
+from scipy.interpolate import interp1d
+
+
+class AvailabilityAnalysis:
+    """
+    A class to perform system availability and loss analysis.
+
+    This class follows the analysis procedure described in [1]_.
+
+    Parameters
+    ----------
+    power_system : pd.Series
+        Timeseries total system power. In the typical case, this is meter
+        power data.
+
+    power_subsystem : pd.DataFrame
+        Timeseries power data, one column per subsystem. In the typical case,
+        this is inverter AC power data. Each column is assumed to represent
+        a subsystem, so no extra columns may be included. The index must
+        match ``power_system``.
+
+    energy_cumulative : pd.Series
+        Timeseries cumulative energy data for the entire system (e.g. meter).
+        These values must be recorded at the device itself (rather than summed
+        by a downstream device like a datalogger or DAS provider) to preserve
+        its integrity across communication interruptions. Units must match
+        ``power`` integrated to hourly energy (e.g. if ``power`` is in kW then
+        ``energy`` must be in kWh).
+
+    power_expected : pd.Series
+        Expected system power data with the same index as the measured data.
+        This can be modeled from on-site weather measurements if there is no
+        risk of instrument calibration or data gaps. However, because full
```

Comment:
Can we rephrase slightly? Maybe "This can be modeled from on-site weather measurements if instruments are well calibrated and there is no risk of data gaps"

Reply:
"no risk of instrument calibration" haha.  What a nightmare it would be if the instruments were calibrated